### ISSUE TITLE
disable directive does not work properly on lines with multiple statements

### ISSUE THREAD

**Issue by smoser** [koalaman/shellcheck#1191]
Attempting a local directive to a line that has multiple statements does not have the desired affect. The combining of 3 statements here on a single line is admittedly not necessary, but is by design to keep it clear that IFS is set back and only changed on that line.

#### For bugs
- My shellcheck version: 0.4.6
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
out="$1"; delim="$2"; orig_ifs="$3"

# shellcheck disable=SC2086
IFS="$delim"; set -- $out; IFS="${orig_ifs}"

# spreading it to multiple lines makes it not complain.
IFS="$delim";
# shellcheck disable=SC2086
set -- $out; IFS="${orig_ifs}"
```

#### Here's what shellcheck currently says:
    Line 4:
    IFS="$delim"; set -- $out; IFS="${orig_ifs}"
                         ^-- SC2086: Double quote to prevent globbing and word splitting.

#### Here's what I wanted or expected to see:
I expect / hoped that the local disable would apply to the unquoted (by design) `set -- $out`.

**Comment by koalaman** [comment]
ShellCheck statements always apply to the immediate single command (simple or compound). This was supposed to make it easy to disable warnings for functions and blocks, but has instead turned out to be a constant source of confusion. It should probably be changed to a line based model at some point.

**Comment by smoser** [comment]
For what its worth, you *can* do this:

    # shellcheck disable=2086
    { IFS=","; set -- $val; IFS="$oifs"; }

Ie, shellcheck does the right thing with the added context {}

**Comment by Dmole** [comment]
Yah I think this is expected behavior, if the behavior were changed how would one prevent it from ignoring things you don't want ignored?

One would need an enable directive.

**Comment by smoser** [comment]
How would you prevent it from ignoring things? Maybe split it into multiple lines?

**Comment by matthewpersico** [comment]
Yes multiple lines would work. In this case, however you might be able to get away with

`IFS=‘;’ set — $val`

Which has the added benefit of not needing the reset. I think. I’ll have to try it after dinner.

**Comment by nicktimko** [comment]
Started using this script and was also pining for some sort of block-disable, which might also help other issues like #1201. My use-case was defining a bunch of exit code bits to later `|` and reduce magic-variableness.

Something like the following?:
```
# shellcheck disable=SC2034 --magic-flag-to-signal-this-persists-vs-applying-to-one-line
FATAL=1
ERROR=2
WARNING=4
# shellcheck enable=SC2034 --or-some-reenabling-flag
```

**Comment by Dmole** [comment]
# shellcheck disable=SC2034
{
FATAL=1
ERROR=2
WARNING=4
}

**Comment by matthewpersico** [comment]
Hello world
```
shell
```
Goodbye world

**Comment by matthewpersico** [comment]
Hello world
```
shell
```
Goodbye world
